### ISSUE TITLE
Form Information not available in template (Umbraco Forms (latest, Umbraco 8)

### ISSUE THREAD

**Issue by thenexus00** [umbraco/Umbraco.Forms.Issues#379]
Referencing: https://github.com/umbraco/Umbraco.Forms.Issues/issues/33

Despite it stating there were features added so you can get FormID etc out in the email templates for Umbraco forms there does not seem to be in data values that work. There is no documentation or updated documentation or intelisense output to show they exist.
There are various forum posts both on later versions of 7 and 8 with people trying but not able to get such values to work. The work arounds indicated in Umbraco 7 accessing form data etc do not work with the changes in Umbraco 8.

### Use case
As there is no custom content support for Umbraco form emails I would like to have content nodes in Umbraco that have a form picker to associate the content to the form.
In the form template I want to get the content node where the form picker GUID matches the current forms GUID used by the template.

### Bug summary

Cant get data out about a form in email templates

### Specifics
At least:
- Want to output form name
- Want to output form guid / ID

**Comment by nul800sebastiaan** [comment]
@thenexus00 both `FormId` and `EntryUniqueId` have been available on `FormsHtmlModel` since Forms v7.0.6, maybe you can show some code of what you're trying to do and where you can't see these properties?

**Comment by ronaldbarendse** [comment]
As commented in the linked issue, the integer form ID isn't available when the email template is rendered, only the GUID: https://github.com/umbraco/Umbraco.Forms.Issues/issues/33#issuecomment-439369278. Using the GUID from the `FormsHtmlModel` and the `IFormService` retrieved from the service container, you can get the form name:

```
@inherits Umbraco.Web.Mvc.UmbracoViewPage<Umbraco.Forms.Core.Models.FormsHtmlModel>
@{
	var form = global::Umbraco.Web.Composing.Current.Factory.GetInstance<Umbraco.Forms.Core.Services.IFormService>().Get(Model.FormId);
	var formName = form.Name;
}
```

**Comment by thenexus00** [comment]
> var form = global::Umbraco.Web.Composing.Current.Factory.GetInstance<Umbraco.Forms.Core.Services.IFormService>().Get(Model.FormId);
> 	var formName = form.Name;

This does not work in current versions of Umbraco forms and returns  Invalid object errors in the console in regard to UFForms

**Comment by alandraper** [comment]
This is crazy. There's no reason you should be calling factories and instantiating services inside razor code. These should be built into the FormsHtmlModel. I would include

- FormName: the name of the form
- URL: the URL where the user completed the form
- SubmittedDate: the date when the form was submitted

**Comment by warrenbuckley** [comment]
Fixed and due in 8.6.2 release
